src/object/README.md

```diff
@@ -114,6 +114,39 @@ and any of the following sharding hints (percentage based on number of targets):
 - DAOS_OCH_SHD_EXT  - max(1024, 80%)
 - DAOS_OCH_SHD_MAX  - 100%
 
+### Object class naming conventions
+
+DAOS object supports two data protection methods: replication(RP) and Erasure Code(EC).
+A set replicated shard, or a set of data and parity shards belonging to the same parity group is
+called redundancy group. An object can be chunked into many redundancy groups in order to achieve
+higher I/O concurrency for better performance and large capactiy because redundancy groups spread
+across different storage engines.
+
+DAOS has over a hundred pre-defined object classes and specific naming conventions for these
+classes:
+- OC: Object Class
+- RP: Replication, the number after underscore is number of replicas. For example: OC_RP_2GX
+  represents two replicas
+- EC: Erasure Code, the number before P is number of data shards, the number after P is number of
+  parity shards in a parity group. For example, OC_EC_4P2G1 represents EC(4+2).
+- G: Redundancy Group, a redundancy group can either be a set of replicated shards or a parity
+  group of EC. The number after G is number of redundancy groups, "X" means the object should
+  spread across as many engines as possible.
+- If there is no RP or EC in the class name, the class has no data protection, in this case the
+  postfix is S{n}, the number after S is number of object shards. similarly, a sharded object can
+  horizontally scale I/O performance and capacity of the object.
+
+### Maximum layout and limitations
+
+DAOS has a few object classes with SX or GX as postfix, for example: OC_SX, OC_RP_2GX. As described
+in the previous section, "X" represents maximum and SX or GX means the object should be placed in
+possibly maximum number of engines in the storage pool.
+
```

Comment:
again, engines is not quite right here. should be targets, which are selected based on `rf_lvl` (?)